cuvarbase/ce.py

```diff
@@ -893,11 +893,24 @@ def batched_run_const_nfreq(self, data, batch_size=10,
 
         [mem.transfer_freqs_to_gpu(**kwargs) for mem in memory]
 
+        best_freqs, best_freq_significances = [], []
+
         for b, batch in enumerate(batches):
             results = self.run(batch, memory=memory, freqs=freqs, **kwargs)
             self.finish()
 
             for i, (f, ce) in enumerate(results):
-                ces.append(np.copy(ce))
-
-        return [(freqs, ce) for ce in ces]
+                ce = np.copy(ce)
+                significance = np.abs(np.mean(ce)-np.min(ce))/np.std(ce)
+                if significance < 1.5:
+                    raise ValueError('CE appears to have memory issues.')
```

Comment:
Unit tests are throwing lots of errors here. @mcoughlin do you know more about the logic here? I'm guessing this is in response to some problems you or others encountered but this seems like an indirect test.

Reply:
Yes definitely remove it. We found that memory issues would lead to basically flat spectra... and then would look fine the next run. It was pretty hard to capture, so we used that trick.

Reply:
Ah yea I vaguely remember this being a problem in my own experience as well...I'll take out this test but will raise an issue if I can reproduce.